lago/plugins/vm.py

```diff
@@ -411,22 +404,35 @@ def copy_to(self, local_path, remote_path, recursive=True):
         with LogTask(
             'Copy %s to %s:%s' % (local_path, self.name(), remote_path),
         ):
-            with self._scp() as scp:
-                scp.put(
-                    files=local_path,
-                    remote_path=remote_path,
-                    recursive=recursive,
-                )
+            try:
+                with self._scp() as scp:
+                    scp.put(
+                        files=local_path,
+                        remote_path=remote_path,
+                        recursive=recursive,
+                    )
+            except OSError as err:
+                raise LagoCopyFilesToVMError(local_path, str(err))
 
     def copy_from(
         self, remote_path, local_path, recursive=True, propagate_fail=True
     ):
-        with self._scp(propagate_fail=propagate_fail) as scp:
-            scp.get(
-                recursive=recursive,
-                remote_path=remote_path,
-                local_path=local_path,
-            )
+        with LogTask(
+            'Copy from %s:%s to %s' % (self.name(), remote_path, local_path),
+        ):
+            try:
+                with self._scp(propagate_fail=propagate_fail) as scp:
+                    scp.get(
+                        recursive=recursive,
+                        remote_path=remote_path,
+                        local_path=local_path,
+                    )
+            except SCPException as err:
+                err_substr = ': No such file or directory'
+                if len(err.args) > 0 and isinstance(
+                    err.args[0], basestring
+                ) and err_substr in err.args[0]:
+                    raise
```

Comment:
Here you should raise `ExtractPathNoPathError` that will tell you that the requested files weren't found.

If thie condition doesn't met, you should raise `LagoCopyFilesFromVMError`

Reply:
Please update the docstrings with `Raises: ...`